bindings/nodejs/src/functions/uart.cc

```diff
@@ -0,0 +1,176 @@
+ * limitations under the License.
+ */
+
+#include <sol-uart.h>
+#include <node.h>
+#include <nan.h>
+
+#include "../common.h"
+#include "../hijack.h"
+#include "../structures/sol-js-uart.h"
+#include "../structures/js-handle.h"
+
+using namespace v8;
+
+class SolUART : public JSHandle<SolUART> {
+public:
+    static const char *jsClassName() { return "SolUART"; }
+};
+
+static void sol_uart_read_callback(void *user_data, struct sol_uart *uart,
+    unsigned char byte_read) {
+    Nan::HandleScope scope;
+    sol_uart_data *uart_data = (sol_uart_data *)user_data;
+    Nan::Callback *callback = uart_data->rx_cb;
+    if (!callback)
+        return;
+
+    Local<Value> arguments[2] = {
+        SolUART::New(uart_data),
+        Nan::New(byte_read)
+    };
+
+    callback->Call(2, arguments);
+}
+
+NAN_METHOD(bind_sol_uart_open) {
+    VALIDATE_ARGUMENT_COUNT(info, 2);
+    VALIDATE_ARGUMENT_TYPE_OR_NULL(info, 0, IsString);
+    VALIDATE_ARGUMENT_TYPE(info, 1, IsObject);
+    sol_uart_config config;
+    sol_uart *uart = NULL;
+
+    if (!hijack_ref())
+        return;
+
+    sol_uart_data *uart_data = new sol_uart_data;
+    uart_data->rx_cb = NULL;
+    uart_data->tx_cb = NULL;
+    if (!c_sol_uart_config(info[1]->ToObject(), uart_data, &config)) {
+        delete uart_data;
+        Nan::ThrowError("Unable to extract sol_uart_config\n");
+        return;
+    }
+
+    Nan::Callback *readCallback = uart_data->rx_cb;
+    config.rx_cb = sol_uart_read_callback;
+
+    uart = sol_uart_open((const char *)*String::Utf8Value(info[0]), &config);
+    if (!uart) {
+        if (readCallback)
+            delete readCallback;
+        delete uart_data;
+        hijack_unref();
+        return;
+    }
+
+    uart_data->uart = uart;
+    info.GetReturnValue().Set(SolUART::New(uart_data));
+}
+
+NAN_METHOD(bind_sol_uart_close) {
+    VALIDATE_ARGUMENT_COUNT(info, 1);
+    VALIDATE_ARGUMENT_TYPE_OR_NULL(info, 0, IsObject);
+    Local<Object> jsUART = Nan::To<Object>(info[0]).ToLocalChecked();
+    sol_uart_data *uart_data = (sol_uart_data *)SolUART::Resolve(jsUART);
+    if (!uart_data)
+        return;
+
+    sol_uart *uart = uart_data->uart;
+    Nan::Callback *callback = uart_data->rx_cb;
+    sol_uart_close(uart);
+    if (callback) {
+        delete callback;
+        delete uart_data;
+        Nan::SetInternalFieldPointer(jsUART, 0, 0);
+        hijack_unref();
+    }
+}
+
+static void sol_uart_write_callback(void *data, struct sol_uart *uart,
+    unsigned char *tx, int status) {
+    Nan::HandleScope scope;
+    Local<Value> buffer;
+    sol_uart_data *uart_data = (sol_uart_data *)data;
+    Nan::Callback *callback = uart_data->tx_cb;
+    if (!callback)
+        return;
+
+    if (status >= 0) {
+        Local <Object> bufObj;
+        bufObj = Nan::NewBuffer((char *)tx, status).ToLocalChecked();
+        buffer = bufObj;
+    } else {
+        buffer = Nan::Null();
+    }
+
+    Local<Value> arguments[3] = {
+        SolUART::New(uart_data),
```

Comment:
Same as https://github.com/solettaproject/soletta/pull/1804/files#r58404435, please use a persistent reference to the ```SolUART``` object you create in ```bind_sol_uart_open()``` and pass it to the callback here instead of creating a new ```SolUART``` object.